File: 백준/Silver/1302. 베스트셀러/베스트셀러.py
import sys
input = sys.stdin.readline

N = int(input())
words = {}

for _ in range(N):
    word = input().strip()

    if word in words:
        words[word] += 1
    else:
        words[word] = 1

max_value = max(words.values())
max_keys = [key for key, value in words.items() if value == max_value]

print(min(max_keys))

File: 백준/Silver/1302. 베스트셀러/베스트셀러.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int N = Integer.parseInt(br.readLine());
        HashMap<String, Integer> words = new HashMap<>();
        
        for(int i = 0; i < N; i++){
            String word = br.readLine().trim();
            words.put(word, words.getOrDefault(word, 0) + 1);
        }
        
        int maxValue = 0;
        for(int value : words.values()) {
            maxValue = Math.max(maxValue, value);
        }
        
        ArrayList<String> S = new ArrayList<>();
        
        for (Map.Entry<String, Integer> e : words.entrySet()) {
            if(e.getValue() == maxValue) {
                S.add(e.getKey());
            }
        }
        Collections.sort(S);
        System.out.print(S.get(0));
    }
}